lint.go

```diff
@@ -1038,14 +1038,17 @@ func (f *file) lintRanges() {
 	})
 }
 
-// lintErrorf examines errors.New calls. It complains if its only argument is an fmt.Sprintf invocation.
+// lintErrorf examines errors.New and testing.Error calls. It complains if its only argument is an fmt.Sprintf invocation.
 func (f *file) lintErrorf() {
 	f.walk(func(node ast.Node) bool {
 		ce, ok := node.(*ast.CallExpr)
 		if !ok {
 			return true
 		}
-		if !isPkgDot(ce.Fun, "errors", "New") || len(ce.Args) != 1 {
+		isErrorsNew := isPkgDot(ce.Fun, "errors", "New")
```

Comment:
This still isn't quite right. It's promoting the particular case too far.

How about something more like this:
```go
check := false
if isPkgDot(ce.Fun, "errors", "New") {
  check = true
} else if se, ok := ce.Fun.(*ast.SelectorExpr); ok {
  check = se.Sel.Name == "Error" && f.pkg.typeOf(se.X).String() == "*testing.T"
}
if !check {
  return true
}
```